mep/common/views.py

```diff
@@ -43,6 +47,56 @@ def dispatch(self, request, *args, **kwargs):
         return response
 
 
+class RdfViewMixin(ContextMixin):
+    '''View mixin to add an RDF linked data graph to context for use in serializing
+    and embedding structured data in templates.'''
+
+    #: default schema.org type for a View
+    rdf_type = SCHEMA_ORG.WebPage
+    #: breadcrumbs, used to render breadcrumb navigation. they should be a list
+    #: of tuples like ('Title', '/url')
+    breadcrumbs = []
+    #: json-ld context for the rdf graph; defaults to schema.org
+    json_ld_context = str(SCHEMA_ORG)
+
+    def get_context_data(self, *args, **kwargs):
+        '''Add generated breadcrumbs and an RDF graph to the view context.'''
+        context = super().get_context_data(*args, **kwargs)
+        context['rdf'] = self.as_rdf().serialize(format='json-ld', auto_compact=True,
+                                                 context=self.json_ld_context, indent=4)
+        context['breadcrumbs'] = self.get_breadcrumbs()
+        return context
+
+    def get_uri(self):
+        '''Get a URI for this page to use for making RDF assertions. Note that
+        this should return a full absolute path, e.g. with absolutize_url().'''
+        raise NotImplementedError
+
+    def as_rdf(self):
+        '''Generate an RDF graph representing the page.'''
+        # add the root node (this page)
+        graph = rdflib.Graph()
+        page_uri = rdflib.URIRef(self.get_uri())
+        graph.add((page_uri, rdflib.RDF.type, self.rdf_type))
+        # generate and add breadcrumbs
```

Comment:
The logic for adding breadcrumbs should probably be conditional on having any returned by `get_breadcrumbs`; I don't know if we'll have any views for this project where we'll have RDF for a page without breadcrumbs, but we might on other projects.